penndata/views.py

```diff
@@ -0,0 +1,203 @@
+        if article:
+            return Response({"article": article})
+        else:
+            return Response({"error": "Site could not be reached or could not be parsed."})
+
+
+class Calendar(APIView):
+    """
+    GET: Returns upcoming university events (within 2 weeks away)
+    """
+
+    def get_events(self):
+        base_url = "https://www.stanza.co/api/schedules/almanacacademiccalendar/"
+        events = []
+        for term in ["fall", "summer", "spring"]:
+            # aggregates events based on term and year
+            url = "{}{}{}term.ics".format(base_url, timezone.localtime().year, term)
+            response = requests.get(url)
+            response.raise_for_status()
+            lines = response.text.split("\n")
+            event = {}
+
+            # adds individual events if and only if it's within 2 weeks from current day
+            for line in lines:
+                if line == "BEGIN:VEVENT":
+                    event = {}
+                elif line.startswith("DTSTART"):
+                    raw_date = line.split(":")[1][0:8]
+                    start_date = datetime.datetime.strptime(raw_date, "%Y%m%d").date()
+                    event["start"] = start_date.strftime("%Y-%m-%d")
+                elif line.startswith("DTEND"):
+                    raw_date = line.split(":")[1][0:8]
+                    end_date = datetime.datetime.strptime(raw_date, "%Y%m%d").date()
+                    event["end"] = end_date.strftime("%Y-%m-%d")
+                elif line.startswith("SUMMARY"):
+                    name = line.split(":")[1]
+                    event["name"] = str(name).strip()
+                elif line == "END:VEVENT":
+                    time_diff = (end_date - timezone.localtime().date()).total_seconds()
+
+                    # checks for within 2 weeks
+                    if time_diff > 0 and time_diff <= 1209600:
+                        # simplifies text
+                        event["name"] = re.split(
+                            "Monday|Tuesday|Wednesday|Thursday|Friday|Saturday|Sunday",
+                            event["name"],
+                        )[0].strip()
+                        event["name"] = re.split(r"\($", event["name"])[0].strip()
+                        event["name"] = event["name"].replace("\\", "")
+                        if "Advance Registration" in event["name"]:
+                            event["name"] = "Advance Registration"
+                        events.append(event)
+
+        events.sort(key=lambda d: d["start"])
+        return events
+
+    def get(self, request):
+        return Response({"calendar": self.get_events()})
+
+
+class Events(generics.ListAPIView):
+    """
+    list:
+    Return a list of events belonging to the type (events/?type=)
+    """
+
+    permission_classes = [AllowAny]
+    serializer_class = EventSerializer
+
+    def get_queryset(self):
+        return Event.objects.filter(event_type=self.kwargs["type"])
```

Comment:
I think this will 500 when `type` isn't provided as a query arg. If that's the case, I think it may be better to do something like `self.kwargs.get("type", "")` so it has a default value